src/main/scala/rv32_5stage/dpath.scala

```diff
@@ -127,9 +137,43 @@ class DatPath(implicit val conf: SodorConfiguration) extends Module
       if_pc_next := if_reg_pc
    }
 
+   // Instruction memory buffer; if the core is stalled and a multi-cycle request arrives, save it in the buffer and supply it to the pipeline once 
+   // the execution is resumed
+   val if_inst_buffer = RegInit(0.U(32.W))
```

Comment:
If this is used in all cores, can it be deduplicated?

Reply:
I believe that this part was only used in 5-stage and 2-stage, and I rewrote the 5-stage instruction buffer with a Chisel Queue. The code are much simpler right now, and I don't think putting into a common frontend would reduce the amount of code. Other cores are having their own fetching buffer that is not compatible with this one.